Unsupported code leads to a long and nasty traceback

Trying to compile this unsupported code
```python
from numpy.random import uniform
# pythran export f()
def f():
    return uniform(-0.5, 0.5)
```
gives this quite nice message
```
CRITICAL: I am in trouble. Your input file does not seem to match Pythran's constraints...
bug_from_numpy_random.py:7:0 error: identifier 'uniform' not found in module 'numpy.random'
----
from numpy.random import uniform
^~~~ (o_0)
----
```

However, with this one
```
from numpy import random
# pythran export f()
def f():
    return random.uniform(-0.5, 0.5)
```
Pythran becomes very angry and crashes with a bad traceback
```
Traceback (most recent call last):
  File "/home/pierre/.pyenv/versions/3.8.2/bin/pythran", line 11, in <module>
    load_entry_point('pythran', 'console_scripts', 'pythran')()
  File "/home/pierre/Dev/pythran/pythran/run.py", line 175, in run
    pythran.compile_pythranfile(args.input_file,
  File "/home/pierre/Dev/pythran/pythran/toolchain.py", line 471, in compile_pythranfile
    output_file = compile_pythrancode(module_name, open(file_path).read(),
  File "/home/pierre/Dev/pythran/pythran/toolchain.py", line 394, in compile_pythrancode
    module, error_checker = generate_cxx(module_name, pythrancode, specs, opts,
  File "/home/pierre/Dev/pythran/pythran/toolchain.py", line 138, in generate_cxx
    pm, ir, docstrings = front_middle_end(module_name, code, optimizations,
  File "/home/pierre/Dev/pythran/pythran/toolchain.py", line 105, in front_middle_end
    refine(pm, ir, optimizations)
  File "/home/pierre/Dev/pythran/pythran/middlend.py", line 43, in refine
    pm.gather(ExtendedSyntaxCheck, node)
  File "/home/pierre/Dev/pythran/pythran/passmanager.py", line 217, in gather
    return a.run(node)
  File "/home/pierre/Dev/pythran/pythran/passmanager.py", line 143, in run
    return super(ModuleAnalysis, self).run(node)
  File "/home/pierre/Dev/pythran/pythran/passmanager.py", line 122, in run
    super(Analysis, self).run(node)
  File "/home/pierre/Dev/pythran/pythran/passmanager.py", line 95, in run
    self.prepare(node)
  File "/home/pierre/Dev/pythran/pythran/passmanager.py", line 90, in prepare
    result = d.run(node)
  File "/home/pierre/Dev/pythran/pythran/passmanager.py", line 143, in run
    return super(ModuleAnalysis, self).run(node)
  File "/home/pierre/Dev/pythran/pythran/passmanager.py", line 122, in run
    super(Analysis, self).run(node)
  File "/home/pierre/Dev/pythran/pythran/passmanager.py", line 96, in run
    return self.visit(node)
  File "/home/pierre/Dev/pythran/pythran/passmanager.py", line 78, in visit
    return super(ContextManager, self).visit(node)
  File "/home/pierre/.pyenv/versions/3.8.2/lib/python3.8/ast.py", line 360, in visit
    return visitor(node)
  File "/home/pierre/.pyenv/versions/3.8.2/lib/python3.8/ast.py", line 368, in generic_visit
    self.visit(item)
  File "/home/pierre/Dev/pythran/pythran/passmanager.py", line 78, in visit
    return super(ContextManager, self).visit(node)
  File "/home/pierre/.pyenv/versions/3.8.2/lib/python3.8/ast.py", line 360, in visit
    return visitor(node)
  File "/home/pierre/Dev/pythran/pythran/analyses/aliases.py", line 543, in visit_FunctionDef
    self.generic_visit(node)
  File "/home/pierre/.pyenv/versions/3.8.2/lib/python3.8/ast.py", line 368, in generic_visit
    self.visit(item)
  File "/home/pierre/Dev/pythran/pythran/passmanager.py", line 78, in visit
    return super(ContextManager, self).visit(node)
  File "/home/pierre/.pyenv/versions/3.8.2/lib/python3.8/ast.py", line 360, in visit
    return visitor(node)
  File "/home/pierre/Dev/pythran/pythran/analyses/aliases.py", line 253, in visit_Return
    ret_aliases = self.visit(node.value)
  File "/home/pierre/Dev/pythran/pythran/passmanager.py", line 78, in visit
    return super(ContextManager, self).visit(node)
  File "/home/pierre/.pyenv/versions/3.8.2/lib/python3.8/ast.py", line 360, in visit
    return visitor(node)
  File "/home/pierre/Dev/pythran/pythran/analyses/aliases.py", line 350, in visit_Call
    self.generic_visit(node)
  File "/home/pierre/.pyenv/versions/3.8.2/lib/python3.8/ast.py", line 370, in generic_visit
    self.visit(value)
  File "/home/pierre/Dev/pythran/pythran/passmanager.py", line 78, in visit
    return super(ContextManager, self).visit(node)
  File "/home/pierre/.pyenv/versions/3.8.2/lib/python3.8/ast.py", line 360, in visit
    return visitor(node)
  File "/home/pierre/Dev/pythran/pythran/analyses/aliases.py", line 378, in visit_Attribute
    return self.add(node, {Aliases.access_path(node)})
  File "/home/pierre/Dev/pythran/pythran/analyses/aliases.py", line 102, in access_path
    return Aliases.access_path(node.value)[demangle(node.attr)]
KeyError: 'uniform'
```
Ok, at least there is 'uniform' at the end...

--- a/pythran/analyses/aliases.py
+++ b/pythran/analyses/aliases.py
@@ -99,7 +99,14 @@ class Aliases(ModuleAnalysis):
         if isinstance(node, ast.Name):
             return MODULES.get(demangle(node.id), node.id)
         elif isinstance(node, ast.Attribute):
-            return Aliases.access_path(node.value)[demangle(node.attr)]
+            attr_key = demangle(node.attr)
+            value_dict = Aliases.access_path(node.value)
+            if attr_key not in value_dict:
+                raise PythranSyntaxError(
+                    "Unsupported attribute '{}' for this object"
+                    .format(attr_key),
+                    node.value)
+            return value_dict[attr_key]
         elif isinstance(node, ast.FunctionDef):
             return node.name
         else:

--- a/pythran/tests/test_import_all.py
+++ b/pythran/tests/test_import_all.py
@@ -64,3 +64,12 @@ class TestImportAll(TestEnv):
 
         with self.assertRaises(pythran.syntax.PythranSyntaxError) as ex:
             pythran.compile_pythrancode("complex_import_manipulation2", dedent(code), pyonly=True)
+
+    def test_complex_import_manipulation3(self):
+        code = """
+            from numpy import random
+            def unsupported_module():
+                return random.i_do_not_exist()"""
+
+        with self.assertRaises(pythran.syntax.PythranSyntaxError) as ex:
+            pythran.compile_pythrancode("complex_import_manipulation3", dedent(code), pyonly=True)

